src/SDKs/Azure.Base/data-plane/Azure.Base/Http/HttpHeader.cs

```diff
@@ -115,63 +38,43 @@ public override bool Equals(object obj)
         }
 
         public bool Equals(HttpHeader other)
-            => _utf8.AsSpan().SequenceEqual(other._utf8);
+        {
+            return string.Equals(Name, other.Name) && Values.Equals(other.Values);
```

Comment:
Should this be invariant comparison?

Reply:
I believe so, as by spec headers are case-insensitive.   

_([RFC 7230, section 3.2](https://tools.ietf.org/html/rfc7230#section-3.2))_